### PR TITLE
FEAT : As OneFM I want to revert transferred responsibilities after Employee's leave duration

### PR DESCRIPTION
**Pull request by samdanikouser**
## Is this a Feature, Chore or Bug?
- [] Feature

## Clearly and concisely describe the feature, chore or bug.
Reassigning the employees responsibility after the employee came from vacation 

## Analysis and design (optional)
[Analyse and attach the design documentation
](https://lucid.app/lucidchart/37a0e91b-1f10-40d0-bb6c-e7126f57eb57/edit?invitationId=inv_b5ee333d-aeee-4565-8774-87103ecbf202&page=lnHdce9qPoSl#)

## Solution description
changes in the reliver_assigment.py file

## Is there a business logic within a doctype?
    - [] No

## Output screenshots (optional)
Screenshot 2024-12-02 at 1 28 00 PM

## Areas affected and ensured
Reliever assignment status - Employee reporter - user permission - projects - todos - operation sites - routine task - simgle doctype - approvals

## Is there any existing behavior change of other features due to this code change?
No.

## Did you test with the following dataset?
- [] Existing Data

## Was child table created?
    - [NO] is attachment required?

## Did you delete custom field?
    - [] No

## Is patch required?
- [] No

## Which browser(s) did you use for testing?
  - [] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/one_fm/doctype/reliever_assignment/reliever_assignment.py [hunk 1]
```diff
@@ -287,4 +286,141 @@ def assign_responsibilities(leave_application):
...
+		frappe.qb.update(ToDo).set(
+				ToDo.allocated_to, self._employee_user_id
+			).where(
+				 (ToDo.name.isin(todos_in_doclist)) 
+   				 & (ToDo.status == "Open")
+			).run()
+		
+	def reassign_roles(self,data):
+		doclist = json.loads(data.doclist)
+		reliver_roles = set(doclist.get("reliever",{}).get("roles",[]))
```
Comment (ks093) [inline one_fm/one_fm/doctype/reliever_assignment/reliever_assignment.py]: @samdanikouser Typo. Please fix reliver_roles to reliever_roles.

File: one_fm/one_fm/doctype/reliever_assignment/reliever_assignment.py [hunk 2]
```diff
@@ -287,4 +286,141 @@ def assign_responsibilities(leave_application):
...
+        frappe.qb.update(Singles)
+        .set(Singles.value, self._employee_user_id)
+        .where((Singles.doctype == record_doc_type) & 
+               (Singles.field == record_field))
+    	).run()
+		frappe.clear_cache(doctype=record_doc_type)
+
+	def reassign(self):
+		leave_application = frappe.get_value("Leave Application", self.leave_application, "name")
+		datas = (frappe.qb.from_("Reliever Assignment Document")\
```
Comment (ks093) [inline one_fm/one_fm/doctype/reliever_assignment/reliever_assignment.py]: @samdanikouser Don't need to get the "datas" again. You already have everything available in self.parent_data.

**Review by ks093 — CHANGES_REQUESTED** [review]

File: one_fm/one_fm/doctype/reliever_assignment/reliever_assignment.py (hunk 2 above)
Comment (samdanikouser) [reply one_fm/one_fm/doctype/reliever_assignment/reliever_assignment.py]: @ks093 Changed as mentioned in the comments and tested the same in local

File: one_fm/one_fm/doctype/reliever_assignment/reliever_assignment.py (hunk 1 above)
Comment (samdanikouser) [reply one_fm/one_fm/doctype/reliever_assignment/reliever_assignment.py]: @ks093 Changed as mentioned in the comments and tested the same in local

**Review by ks093 — APPROVED** [review]

**Pull request merged**